src/appleseed/foundation/platform/path.cpp

```diff
@@ -133,6 +135,49 @@ const char* get_executable_directory()
     return path;
 }
 
+const char* get_home_directory()
+{
+    static char path[FOUNDATION_MAX_PATH_LENGTH + 1];
+    static bool path_initialized = false;
+
+    if (!path_initialized)
+    {
+#if defined _WIN32
+
+        return 0;
+
++#elif defined __APPLE__
+
+        return 0;
+
+#elif defined __linux__
+
+        const char *home_dir = getenv("HOME");
+
+        if (home_dir == 0)
+            home_dir = getpwuid(getuid())->pw_dir;
+
+        assert(home_dir);
+        strcpy(path, home_dir);
```

Comment:
@est77 yes, this code works identically on Linux/glibc and FreeBSD; complete version with required headers:
```c++
#include <sys/types.h>  // not really required (we don't declare any uid_t, etc. variables)
#include <stdio.h>
#include <stdlib.h>
#include <pwd.h>
#include <unistd.h>

int main()
{
    const char *home_dir = getenv("HOME2"); // HOME2 to emulate missing variable

    if (home_dir == 0)
        home_dir = getpwuid(getuid())->pw_dir;

    printf("my sweet home: %s\n", home_dir);
    return 0;
}
```